src/main/java/org/spongepowered/mod/mixin/world/MixinWorld.java

```diff
@@ -93,21 +125,106 @@ public BlockLoc getBlock(int x, int y, int z) {
 
     @Override
     public Biome getBiome(Vector3d position) {
-        throw new UnsupportedOperationException();
+        BlockPos blockPos = new BlockPos(position.getX(), position.getY(), position.getZ());
+        return new SpongeBiome(blockPos, getBiomeGenForCoords(blockPos), getChunkFromBlockCoords(blockPos));
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public Collection<Entity> getEntities() {
-        return new ArrayList<Entity>();
+        return (Collection<Entity>) (Object) loadedEntityList;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Collection<Entity> getEntities(Predicate<Entity> filter) {
+        return (Collection<Entity>) (Object) func_175644_a(net.minecraft.entity.Entity.class,
+                (Predicate<net.minecraft.entity.Entity>) (Object) filter);
     }
 
     @Override
     public Optional<Entity> createEntity(EntityType type, Vector3d position) {
+        try {
+            net.minecraft.entity.Entity entity =
+                    (net.minecraft.entity.Entity) type.getEntityClass().getConstructor(net.minecraft.world.World.class).newInstance(this);
+            entity.setPosition(position.getX(), position.getY(), position.getZ());
+            if (spawnEntityInWorld(entity)) {
+                return Optional.of((Entity) entity);
+            }
+        } catch (Exception ignored) {
+            // Entity constructor does not have single World parameter, too bad
+        }
         return Optional.absent();
     }
 
     @Override
+    public Optional<Entity> createEntity(EntitySnapshot snapshot, Vector3d position) {
+        return createEntity(snapshot.getType(), position);
+    }
+
+    @Override
     public WorldBorder getWorldBorder() {
         return (WorldBorder) shadow$getWorldBorder();
     }
+
+    @Override
+    public void playSound(Sound sound, Vector3d position, double volume) {
+        playSound(sound, position, volume, 1); // Is 1 the default pitch?
+    }
+
+    @Override
+    public void playSound(Sound sound, Vector3d position, double volume, double pitch) {
+        playSound(sound, position, volume, pitch, 0);
+    }
+
+    @Override
+    public void playSound(Sound sound, Vector3d position, double volume, double pitch, double minVolume) {
+        playSoundEffect(position.getX(), position.getY(), position.getZ(), sound.getName(), (float) Math.max(minVolume, volume), (float) pitch);
+    }
+
+    @Override
+    public Optional<Entity> getEntityFromUUID(UUID uuid) {
+        for (net.minecraft.entity.Entity entity : this.loadedEntityList) {
+            if (entity.getUniqueID().equals(uuid)) {
+                return Optional.of((Entity) entity);
```

Comment:
There is a better way to do this, WorldServer holds a Map of 'entitiesByUuid' which you can use to directly lookup the entity.